### PR TITLE
0.9 authorise->authorize, fix param order

### PR DESCRIPTION
**Pull request by trenouf**
I think we haven't decided yet about whether to fix param order in 0.8, but in the meantime one of the two commits here fixes it in 0.9.

### CODE REVIEW

**Comment by mattheworiordan** [comment]
I would like to see `(RSA10l) Has an alias method RestClient#authorise and RealtimeClient#authorise that will log a deprecation warning stating that this alias method will be removed in v1.0 and the user should instead use authorize` implemented with this if possible.  Should be straightforward.  The question then is do we use the old authorise signature or the new correct one, clearly I prefer using the new one to avoid all of this unnecessary confusion.

File: lib/src/main/java/io/ably/lib/rest/Auth.java [hunk 1]
```diff
@@ -438,10 +438,18 @@ public TokenDetails authorise(TokenParams params, AuthOptions options) throws Ab
 	}
 
 	/**
+	 * Alias of authorize() (0.9 RSA10l)
+	 */
+	public TokenDetails authorise(TokenParams params, AuthOptions options) throws AblyException {
+		Log.w(TAG, "authorise() alias will be removed in 1.0");
```
Comment (mattheworiordan) [inline lib/src/main/java/io/ably/lib/rest/Auth.java]: Oh, sorry, not sure how I missed that :/

**Comment by trenouf** [comment]
So is this approved? Can I merge it?

File: lib/src/main/java/io/ably/lib/http/TokenAuth.java [hunk 2]
```diff
@@ -36,21 +36,21 @@ public void setTokenDetails(TokenDetails tokenDetails) {
 		this.encodedToken = Base64Coder.encodeString(tokenDetails.token).replace("=", "");
 	}
 
-	public TokenDetails authorise(AuthOptions options, TokenParams params) throws AblyException {
-		Log.i("TokenAuth.authorise()", "");
+	public TokenDetails authorize(AuthOptions options, TokenParams params) throws AblyException {
```
Comment (paddybyers) [inline lib/src/main/java/io/ably/lib/http/TokenAuth.java]: I realise the spec doesn't require us to change this, but why not do it and keep things consistent?

**Comment by paddybyers** [comment]
> So is this approved? Can I merge it?

Sorry, pls see my comment.

File: lib/src/main/java/io/ably/lib/http/TokenAuth.java (hunk 2 above)
Comment (trenouf) [inline lib/src/main/java/io/ably/lib/http/TokenAuth.java]: @paddy You mean the param order?

**Comment by paddybyers** [comment]
Thanks

**Pull request merged**